src/ForeignController.sol

```diff
@@ -117,27 +140,48 @@ contract ForeignController is AccessControl {
             abi.encodeCall(usdc.approve, (address(cctp), usdcAmount))
         );
 
-        // Send USDC to CCTP for bridging to destinationDomain
-        proxy.doCall(
-            address(cctp),
-            abi.encodeCall(
-                cctp.depositForBurn,
-                (
-                    usdcAmount,
-                    destinationDomain,
-                    mintRecipient,
-                    address(usdc)
+        // If amount is larger than limit we must break it up
+        uint256 burnLimit = cctp.localMinter().burnLimitsPerMessage(address(usdc));
+        while (usdcAmount > burnLimit) {
+            proxy.doCall(
+                address(cctp),
+                abi.encodeCall(
+                    cctp.depositForBurn,
+                    (
+                        burnLimit,
+                        destinationDomain,
+                        mintRecipient,
+                        address(usdc)
+                    )
                 )
-            )
-        );
+            );
+
+            usdcAmount -= burnLimit;
+        }
+
+        // Send remainder if any
+        if (usdcAmount > 0) {
+            proxy.doCall(
+                address(cctp),
+                abi.encodeCall(
+                    cctp.depositForBurn,
+                    (
+                        usdcAmount,
+                        destinationDomain,
+                        mintRecipient,
+                        address(usdc)
+                    )
+                )
+            );
```

Comment:
I'd extract it to sth like `function initiateCCTPTransfer(uint256) internal`

Reply:
Good call. Done.